README.md

```diff
@@ -76,8 +76,48 @@ Generally speaking, build as follows:
 `make check` runs only sysadmin's tests. If you wish to run the decibel-cpp tests, run
 `make decibel-check`.
 
-If this is your first time, you might have some dependencies which need to be installed
-first.
+Dockerized Development
+======================
+
+For consistancy it's recommended to build and test in Docker. We will try an up to date image
+published, but if it's not on Dockerhub you can build it yourself. 
+
+The docker commands are wrapped in the docker_control.sh script. 
+
+```bash
+ #  make sure you have submodules
+git submodule update --init --recursive
+
+ # Attempt to pull the dependency container. If that fails 
+ you will build it yourself. 
+./docker_control.sh -b
+
+
+```
+
+This make an image that includes all the 3rd party dependencies. To build Sysadmin
+inside the container you will start a container with the code mounted in.
+Note that you run as a non-root user with your host UID inside the container
+so you shouldn't have to worry about permissions on files you create.
+
+```bash
+# build and test in a single command. Non-interactive 
+./docker_control.sh -t
+
+
+# Run the container interactively 
+./docker_control.sh -i
+
+# while in interactive mode you can manually build and test.
+mkdir -p build
```

Comment:
This feels redundant. You could just make a point about how you can follow the typical development instructions in the interactive mode.